modules/tinymce/src/plugins/textpattern/test/ts/browser/TrailingPunctuationTest.ts

```diff
@@ -1,75 +1,53 @@
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'textpattern',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ], true);
 
-    TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-      const tinyApis = TinyApis(editor);
-      const tnt = sTypeAndTrigger(tinyApis, editor);
+  const pTypeAndTriggerTest = (patternText: string, trigger: string, tag: string, rawText: string) => async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>' + patternText + trigger + '</p>');
+    TinySelections.setCursor(editor, [ 0, 0 ], patternText.length + 1);
+    TinyContentActions.keypress(editor, trigger.charCodeAt(0));
+    await Waiter.pTryUntil(
+      'did not get expected format',
+      () => TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str) => {
+        return s.element('body', {
+          children: [
+            s.element('p', {
+              children: [
+                s.element(tag, {
+                  children: [
+                    s.text(str.is(rawText))
+                  ]
+                }),
+                s.text(str.is(trigger), true)
+              ]
+            })
+          ]
+        });
+      }))
+    );
+  };
 
-      Pipeline.async({}, Log.steps('TBA', 'TextPattern: TrailingPunctuationTest', [
-        tnt('em with ,', '*a*', ',', 'em', 'a'),
-        tnt('strong with ,', '**a**', ',', 'strong', 'a'),
-        tnt('em with .', '*a*', '.', 'em', 'a'),
-        tnt('strong with .', '**a**', '.', 'strong', 'a'),
-        tnt('em with ;', '*a*', ';', 'em', 'a'),
-        tnt('strong with ;', '**a**', ';', 'strong', 'a'),
-        tnt('em with :', '*a*', ':', 'em', 'a'),
-        tnt('strong with :', '**a**', ':', 'strong', 'a'),
-        tnt('em with !', '*a*', '!', 'em', 'a'),
-        tnt('strong with !', '**a**', '!', 'strong', 'a'),
-        tnt('em with ?', '*a*', '?', 'em', 'a'),
-        tnt('strong with ?', '**a**', '?', 'strong', 'a')
-      ]), onSuccess, onFailure);
-    }, {
-      plugins: 'textpattern',
-      base_url: '/project/tinymce/js/tinymce'
-    }, success, failure);
-  }
-);
+  it('em with ,', pTypeAndTriggerTest('*a*', ',', 'em', 'a'));
+  it('strong with ,', pTypeAndTriggerTest('**a**', ',', 'strong', 'a'));
+  it('em with .', pTypeAndTriggerTest('*a*', '.', 'em', 'a'));
+  it('strong with .', pTypeAndTriggerTest('**a**', '.', 'strong', 'a'));
+  it('em with ;', pTypeAndTriggerTest('*a*', ';', 'em', 'a'));
+  it('strong with ;', pTypeAndTriggerTest('**a**', ';', 'strong', 'a'));
+  it('em with :', pTypeAndTriggerTest('*a*', ':', 'em', 'a'));
+  it('strong with :', pTypeAndTriggerTest('**a**', ':', 'strong', 'a'));
+  it('em with !', pTypeAndTriggerTest('*a*', '!', 'em', 'a'));
+  it('strong with !', pTypeAndTriggerTest('**a**', '!', 'strong', 'a'));
+  it('em with ?', pTypeAndTriggerTest('*a*', '?', 'em', 'a'));
+  it('strong with ?', pTypeAndTriggerTest('**a**', '?', 'strong', 'a'));
```

Comment:
Out of scope but this could probably be split into two contexts? I dunno

Reply:
Have broken it up into 2 contexts.